crates/iota-sdk-types/src/lib.rs

```diff
@@ -201,6 +202,50 @@ macro_rules! impl_uniffi_byte_vec_wrapper {
     ($id:ident) => {};
 }
 
+#[macro_export]
+macro_rules! def_is {
+    ($($variant:ident),* $(,)?) => {
+        paste::paste! {$(
+        #[inline]
+        pub fn [< is_ $variant:snake >](&self) -> bool {
+            matches!(self, Self::$variant)
+        }
+        )*}
+    };
+}
+
+#[macro_export]
+macro_rules! def_is_as_opt {
+    ($($variant:ident => $inner:ty),* $(,)?) => {
+        paste::paste! {$(
+        #[inline]
+        pub fn [< is_ $variant:snake >](&self) -> bool {
+            matches!(self, Self::$variant(_))
+        }
+
+        #[inline]
+        pub fn [< as_ $variant:snake >](&self) -> &$inner {
+            let Self::$variant(inner) = self else {
+                panic!("not a {}", stringify!($variant));
+            };
+            inner
+        }
+
+        #[inline]
+        pub fn [< as_ $variant:snake _opt >](&self) -> Option<&$inner> {
+            if let Self::$variant(inner) = self {
+                Some(inner)
+            } else {
+                None
+            }
+        }
+        )*}
+    };
+    ($($variant:ident),* $(,)?) => {
+        crate::def_is_as_opt!{$($variant => $variant,)*}
+    };
+}
+
```

Comment:
Are these macros now needed only for uniffi? Should be in the uniffi_helpers.rs maybe?